### PR TITLE
feat: add migration mode option to `Install-EsxiCertificate`

### LINKED ISSUE #32: Add support for ESXi certificate replacement without performing full evacuation when entering maintenance mode
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Currently, when using vSAN, it is required to perform full evacuation of VMs when putting the ESXi host in maintenance mode. 

Add support for ESXi certificate replacement without performing full evacuation when entering maintenance mode.

### Use Case(s)

NA

### Potential Configuration

NA

### References

_No response_

### LINKED ISSUE #100: when calling Install-EsxiCertificate with esxiFqdn will not enter maintenance mode properly
### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-certificate-management/blob/main/CODE_OF_CONDUCT.md).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

5.1

### PowerShell Version

PSVersion 7.3.7 Core

### PowerCLI Version

13.1.0

### Module Version

1.4.0

### PowerValidatedSolutions Version

2.7.1

### PowerVCF Version

2.4.0

### Guest Operating System

windows server 2019

### Environment Details

_No response_

### Description

When calling Install-EsxiCertificate with -esxiFqdn, the host will not migrate any vSAN data off to other hosts.  

### Error or Debug Output

https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-certificate-management/blob/7bad4f79a246edb117554f0826f129e24717770f/VMware.CloudFoundation.CertificateManagement.psm1#L1128C30-L1128C30

$clusterDetails is empty due to $cluster parameter is no passed when using esxiFqdn

### Expected Behavior

ESXi host will migrate any power on VMs to other hosts using either full or EnsureAccessibility

### Actual Behavior

ESXi host enters maintenance mode immediately and making all power on vms on the ESXi host into inaccessible state

### Steps to Reproduce

Install-EsxiCertificate -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -esxiFqdn sfo01-m01-esx01.sfo.rainpole.io -certificateDirectory F:\certificates -certificateFileExt ".cer"

### Log Fragments and Files

_No response_

### Screenshots

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by garlicNova**
### Summary

- Updated `Install-EsxiCertificate` to include the `vsanDataMigrationMode` parameter. The ESXi host will enter maintenance mode with Migrate Power off VMs option disabled and vSAN data migration ,ode set to `EnsureAccessibility`.

- Add `Get-EsxiHostVsanMaintenanceModePrecheck` function to check for any issue with ESXi host entering a particular vSAN maintenance mode.

### Type

- [x] Bugfix
- [x] Enhancement or Feature
- [ ] Code Style or Formatting
- [x] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [x] Yes, there are breaking changes.
- [ ] No, there are no breaking changes.

### Test and Documentation

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

### Issue References

Closes #32 
Closes #100

### CODE REVIEW

File: docs/documentation/functions/Get-EsxiHostVsanMaintenanceModePrecheck.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+# Get-EsxiHostVsanMaintenanceModePrecheck
+
+## Synopsis
+
+Pre check for any issues when the ESXi Host enters a particular vSAN maintenance mode.
+
+## Syntax
+
+```powershell
+Get-EsxiLockdownMode [-server] <String> [-user] <String> [-pass] <String> [-domain] <String> [-cluster] <String> [[-esxiFqdn] <String>] [<CommonParameters>]
```
Comment (bhumitra) [inline docs/documentation/functions/Get-EsxiHostVsanMaintenanceModePrecheck.md:10]: Update the example. It should be Get-EsxiHostVsanMaintenanceModePrecheck

**Comment by bhumitra** [comment]
@garlicNova The `Install-VCFCertificate` cmdlet is the one which is exposed to the user. 
Hence for the user to use the changes made for `Install-EsxiCertificate` for ensureAccesibility, we must allow configuration of the parameters from Install-VCFCertificate. Could you please make that change.

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
@garlicNova - can you address Bhumitra's comments today?

File: VMware.CloudFoundation.CertificateManagement.psd1 [hunk 2]
```diff
@@ -11,7 +11,7 @@
     RootModule = '.\VMware.CloudFoundation.CertificateManagement.psm1'
 
     # Version number of this module.
-    ModuleVersion = '1.5.0.1003'
+    ModuleVersion = '1.6.0.1000'
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.CertificateManagement.psd1:14]: Fixed the version.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
 
+## v1.6.0
+
+> Release Date: Unreleased
+
+Enhancement:
+
+- Added `sanDataMigrationMode` option to `Install-EsxiCertificate` cmdlet. Options include `Full` and `EnsureAccessibility`. [GH-101](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-certificate-management/pull/101)
+- Added `Get-EsxiHostVsanMaintenanceModePrecheck` cmdlet. [GH-101](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-certificate-management/pull/101)
+
```
Comment (tenthirtyam) [inline CHANGELOG.md:11]: Updated.

File: VMware.CloudFoundation.CertificateManagement.psm1 [hunk 4]
```diff
@@ -734,6 +734,95 @@ Function Get-EsxiConnectionState {
...
 }
 
+Function Get-EsxiHostVsanMaintenanceModePrecheck {
+    <#
+        .SYNOPSIS
+        Checks for any issues when the ESXi H=host enters a particular vSAN maintenance mode.
+
+        .DESCRIPTION
+        The Get-EsxiHostVsanMaintenanceModePrecheck cmdlet checks if there's any issues for the ESXi host entering a particular vSAN maintenance mode.
+        The cmdlet will halt the script if the pre check fails.
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.CertificateManagement.psm1:744]: Updated.

File: docs/documentation/functions/Get-EsxiHostVsanMaintenanceModePrecheck.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
+# Get-EsxiHostVsanMaintenanceModePrecheck
+
+## Synopsis
+
+Checks for any issues when the ESXi H=host enters a particular vSAN maintenance mode.
```
Comment (tenthirtyam) [inline docs/documentation/functions/Get-EsxiHostVsanMaintenanceModePrecheck.md:5]: Updated.

File: VMware.CloudFoundation.CertificateManagement.psm1 [hunk 6]
```diff
@@ -1962,12 +2101,14 @@ Function Install-VCFCertificate {
...
+        will enter maintenance mode with Migrate Power off VMs option enabled and vSAN data migration Mode set to EnsureAccessibility.
 
         .EXAMPLE
-        Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -cluster sfo-m01-cl01 -certificateDirectory F:\certificates -certificateFileExt ".cer"
-        This example will install certificates for each ESXi host in cluster sfo-m01-cl01 in workload domain sfo-m01 from the provided path.
+        Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -cluster sfo-m01-cl01 -vsanDataMigrationMode EnsureAccessibility -certificateDirectory F:\certificates -certificateFileExt ".cer"
+        This example will install certificates for each ESXi host in cluster sfo-m01-cl01 in workload domain sfo-m01 from the provided path.  The ESXi host 
+        will enter maintenance mode with Migrate Power off VMs option disabled and vSAN data migration Mode set to EnsureAccessibility.
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.CertificateManagement.psm1:2111]: Updated.

File: VMware.CloudFoundation.CertificateManagement.psm1 [hunk 7]
```diff
@@ -1962,12 +2101,14 @@ Function Install-VCFCertificate {
         This example will connect to SDDC Manager to install the signed certificates for a given workload domain.
 
         .EXAMPLE
-        Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -esxiFqdn sfo01-m01-esx01.sfo.rainpole.io -certificateDirectory F:\certificates -certificateFileExt ".cer"
-        This example will install the certificate to the ESXi host sfo01-m01-esx01.sfo.rainpole.io in domain sfo-m01 from the provided path.
+        Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -esxiFqdn sfo01-m01-esx01.sfo.rainpole.io -migratePowerOffVMs -vsanDataMigrationMode EnsureAccessibility -certificateDirectory F:\certificates -certificateFileExt ".cer"
+        This example will install the certificate to the ESXi host sfo01-m01-esx01.sfo.rainpole.io in domain sfo-m01 from the provided path.  The ESXi host 
+        will enter maintenance mode with Migrate Power off VMs option enabled and vSAN data migration Mode set to EnsureAccessibility.
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.CertificateManagement.psm1:2106]: Updated.

File: docs/documentation/functions/Install-VCFCertificate.md [hunk 8]
```diff
@@ -49,18 +49,18 @@ This example will connect to SDDC Manager to install the signed certificates for
 ### Example 2
 
 ```powershell
-Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -esxiFqdn sfo01-m01-esx01.sfo.rainpole.io -certificateDirectory F:\certificates -certificateFileExt ".cer"
+Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -esxiFqdn sfo01-m01-esx01.sfo.rainpole.io -migratePowerOffVMs -vsanDataMigrationMode EnsureAccessibility -certificateDirectory F:\certificates -certificateFileExt ".cer"
 ```
 
-This example will install the certificate to the ESXi host sfo01-m01-esx01.sfo.rainpole.io in domain sfo-m01 from the provided path.
+This example will install the certificate to the ESXi host sfo01-m01-esx01.sfo.rainpole.io in domain sfo-m01 from the provided path. The ESXi host will enter maintenance mode with Migrate Power off VMs option enabled and vSAN data migration Mode set to `EnsureAccessibility`.
```
Comment (tenthirtyam) [inline docs/documentation/functions/Install-VCFCertificate.md:55]: Updated.

File: docs/documentation/functions/Install-VCFCertificate.md [hunk 9]
```diff
@@ -49,18 +49,18 @@ This example will connect to SDDC Manager to install the signed certificates for
...
 
 ### Example 3
 
 ```powershell
-Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user <administrator@vsphere.local> -pass VMw@re1! -domain sfo-m01 -cluster sfo-m01-cl01 -certificateDirectory F:\certificates -certificateFileExt ".cer"
+Install-VCFCertificate -esxi -server sfo-vcf01.sfo.rainpole.io -user <administrator@vsphere.local> -pass VMw@re1! -domain sfo-m01 -cluster sfo-m01-cl01 -vsanDataMigrationMode EnsureAccessibility -certificateDirectory F:\certificates -certificateFileExt ".cer"
 ```
 
-This example will install certificates for each ESXi host in cluster sfo-m01-cl01 in workload domain sfo-m01 from the provided path.
+This example will install certificates for each ESXi host in cluster sfo-m01-cl01 in workload domain sfo-m01 from the provided path. The ESXi host will enter maintenance mode with Migrate Power off VMs option disabled and vSAN data migration Mode set to `EnsureAccessibility`.
```
Comment (tenthirtyam) [inline docs/documentation/functions/Install-VCFCertificate.md:63]: Updated.

**Review by tenthirtyam — APPROVED** [review]
Made some updates to quicken the review.

Now LGTM!

**Comment by tenthirtyam** [comment]
Passing to @bhumitra for next review.

File: docs/documentation/functions/Get-EsxiHostVsanMaintenanceModePrecheck.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+Required: True
+Position: 2
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -pass
+
+The passwordto authenticate to the SDDC Manager instance.
```
Comment (bhumitra) [inline docs/documentation/functions/Get-EsxiHostVsanMaintenanceModePrecheck.md:73]:
```suggestion
The password to authenticate to the SDDC Manager instance.
```

**Review by bhumitra — DISMISSED** [review]
I am good with the changes now. Made a minor fix. 🚀  LGTM

**Comment by tenthirtyam** [comment]
Thanks! I'll rebase and merge.

**Pull request merged**